### DISCUSSION TITLE
How to use all the available GPUs

### DISCUSSION THREAD

**Discussion by mnslarcher** [Lightning-AI/pytorch-lightning#12755] [DDP / multi-GPU / multi-node] [answered]
Hi all, I migrated my code to the latest version of lightnining and tried using `accelerator=auto` and `devices=-1` to use all available GPUs but I get:

UserWarning: The flag `devices=-1` will be ignored, instead the device specific number 1

What is the recommended way to use, by default, all GPUs (when present)?

**Comment by akihironitta** [comment]
Hi @mnslarcher! To use all of available devices, use:
```python
Trainer(..., devices="auto")
```

The doc page is available here: https://pytorch-lightning.readthedocs.io/en/latest/common/trainer.html#devices

**Reply by mnslarcher** [reply]
Thanks @akihironitta , this is what I did, I was thinking that the the "-1" was wrong but then I check again the doc and found that what I did was exactly what the doc (and you) suggest, I then opened this issue: https://github.com/PyTorchLightning/pytorch-lightning/issues/12756

**Comment by akihironitta** [comment, marked as answer]
For anyone seeing this discussion, have a look at https://github.com/Lightning-AI/lightning/issues/12756#issuecomment-1106629824.
